### PR TITLE
PE-81 | Create a Service to Send an Answer to Backend

### PR DESCRIPTION
**Pull request by dominikyz**
What I Did:

- Created a service that sends answer to MongoDB

How to Test:

- Switch to branch 81
- Run : `git fetch` and  `git pull`
- On the home page, click on any question's `More details`.
- You should see `Answer button`. Click on it and select `No chat`. Type your answer and submit.
- Check MongoDB Answers and make sure you see your answer entry there.

image

### CODE REVIEW

File: lib/services/new_answer_service.dart [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    final response = await http.post(
+      Uri.http(
+        DotEnv.env['HOST'] + ':' + DotEnv.env['PORT'],
+        '/v1/answers',
+      ),
+      headers: <String, String>{
+        'Content-type': 'application/json; charset=UTF-8',
+      },
+      body: jsonEncode(
+        answer,
```
Comment (tonynguyen98) [inline lib/services/new_answer_service.dart:20]: I'm actually surprised that you didn't need to call `.toJson` here, but it seems to be working...

File: lib/screens/new_form_screens/new_form.dart [hunk 2]
```diff
@@ -14,17 +14,20 @@ import 'package:firebase_storage/firebase_storage.dart';
 import 'package:project_eureka_flutter/services/email_auth.dart';
 import 'package:uuid/uuid.dart';
 import 'package:project_eureka_flutter/services/new_question_service.dart';
+import 'package:project_eureka_flutter/services/new_answer_service.dart';
 
 class NewForm extends StatefulWidget {
   final String categoryName;
   final bool isAnswer;
   final QuestionModel questionModel;
```
Comment (tonynguyen98) [inline lib/screens/new_form_screens/new_form.dart]: You can remove this `questionModel` If you're just gonna use the String `questionId` instead. This is actually a better workflow.

**Review by tonynguyen98 — APPROVED** [review]
Working as intended! Just some minor suggestions.

File: lib/services/new_answer_service.dart (hunk 1 above)
Comment (dominikyz) [reply lib/services/new_answer_service.dart:20]: i'll leave to json function, just in case we need it in the future

**Review by davidsquines — APPROVED** [review]
Works as intended

**Review by txt020 — APPROVED** [review]
@dominikyz Everything looks good, but the video button is not implemented. I am assuming this is for a future build.

**Pull request merged**